oneview_redfish_toolkit/services/zone_service.py

```diff
@@ -97,16 +99,40 @@ def get_zone_ids_by_templates(self, server_profile_templates):
                 server_profile_templates: the list of Server Profile Template
         """
         zone_ids = []
+        valid_logical_enclosures = self._get_valid_logical_enclosures()
         for template in server_profile_templates:
             template_id = template["uri"].split("/")[-1]
             controller = ComputerSystemService.get_storage_controller(template)
             if controller:
-                enclosures_uris = self._get_enclosures_uris_by_template(
+                enclosures_uris_by_spt = self._get_enclosures_uris_by_template(
                     template)
-                for encl_uri in enclosures_uris:
+                enclosures_uris = set(enclosures_uris_by_spt)\
+                    .intersection(valid_logical_enclosures)
+
+                for encl_uri in sorted(enclosures_uris):
                     zone_id = ZoneService.build_zone_id(template_id, encl_uri)
                     zone_ids.append(zone_id)
             else:
                 zone_ids.append(template_id)
 
         return zone_ids
+
+    def _get_valid_logical_enclosures(self):
+        all_logical_enclosures_uris = self._get_all_enclosures_uris()
+        valid_logical_enclosures = list()
+        for logical_enclosure_uri in all_logical_enclosures_uris:
+            logical_encl_has_drives = self.ov_client.drive_enclosures.get_all(
```

Comment:
In another [point of project](https://github.com/HewlettPackard/oneview-redfish-toolkit/blob/master/oneview_redfish_toolkit/blueprints/zone.py#L75-L78) we are getting information about drive enclosures associated to the enclosure, too.
We could verify what is the best way to this current case, because the result here comes with all drives too, resulting to a possible very large data returned and we just want verify if drive enclosure is present.
Could you verify the best approach, please? Let me know about it. 😃

Reply:
The approach suggested by you returns more data that I will use. The request that I do only returns a 'members' list that is the drive enclosures associated or not with the enclosure filtered. If there's no drive enclosure, the list will be empty. So it is more simple to check the data in this case than in yours. But I appreciate your suggestion 👍